app/form_models/jobseekers/job_preferences_form.rb

```diff
@@ -128,10 +128,19 @@ class LocationForm
       attribute :radius
 
       validates :location, :radius, presence: true
+      validate :location_within_united_kingdom
 
       def radius_options
         [0, 1, 5, 10, 15, 20, 25, 50, 100, 200].map { |radius| [radius, I18n.t("jobs.search.number_of_miles", count: radius)] }
       end
+
+      private
+
+      def location_within_united_kingdom
+        return if Geocoder.search(location).map(&:country).include?("United Kingdom")
+
+        errors.add(:location, I18n.t("activemodel.errors.models.jobseekers/job_preferences_form/location_form.attributes.location.blank"))
+      end
```

Comment:
Suggestion :raising_hand: 

If I did read this right, both custom validation methods in the `JobPreferencesForm` and the `SubscriptionForm` do the same and present the same validation error.  

Have you considered extracting this as a custom Validator? It looks like I would be a good fit for it.

EG:
```ruby
class WithinUnitedKingdomValidator< ActiveModel::EachValidator
  def validate_each(record, attribute, value)
    return if value.blank?

    unless Geocoder.search(value).map(&:country).include?("United Kingdom")
      record.errors.add attribute, (blablabla)
    end
  end
end

class JobPreferencesForm
  ...
  validates :location, :radius, presence: true, within_united_kingdom: true
  ...
end
```